samples/Rewrite/FileParser/RewriteConfigurationFileParser.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Rewrite.Structure2;
+using Rewrite.ConditionParser;
+using Rewrite.RuleParser;
+namespace Rewrite.FileParser
+{
+    public static class RewriteConfigurationFileParser
+    {
+        private static StreamReader _reader;
+        public static List<Rule> Parse(Stream input)
```

Comment:
If I call this from two different threads I break your code :) You shouldn't keep a static reference to _reader because the second call will overwrite it